internal/connector/internal/workers/connector_type_mgr.go

```diff
@@ -0,0 +1,179 @@
+			Reconciler: reconciler,
+		},
+		connectorClusterService: connectorClusterService,
+		connectorTypesService:   connectorTypesService,
+		startupReconcileDone:    false,
+		startupReconcileStopped: false,
+		env:                     env,
+	}
+
+	// The release of this waiting group signal the http service to start serving request
+	// this needs to be done across multiple instances of fleetmanager running,
+	// and yet just one instance of ConnectorTypeManager of those multiple fleet manager will run the reconcile loop.
+	// The release of the waiting group must then be done outside the reconcile loop,
+	// the condition is checked in runStartupReconcileCheckWorker().
+	result.startupReconcileWG.Add(1)
+
+	// Mark startupReconcileWG as done in a separate goroutine instead of in worker reconcile
+	// this is required to allow multiple instances of fleetmanager to startup.
+	result.runStartupReconcileCheckWorker()
+	return result
+}
+
+// Start initializes the connector manager to reconcile connector requests
+func (k *ConnectorTypeManager) Start() {
+	k.StartWorker(k)
+}
+
+// Stop causes the process for reconciling connector requests to stop.
+func (k *ConnectorTypeManager) Stop() {
+	k.StopWorker(k)
+}
+
+func (k *ConnectorTypeManager) Reconcile() []error {
+	if !k.startupReconcileDone {
+		glog.V(5).Infoln("Reconciling startup connector catalog updates...")
+
+		// the assumption here is that this runs on one instance only of fleetmanager,
+		// runs only at startup and while requests are not being served
+		if err := k.connectorTypesService.DeleteUnusedAndNotInCatalog(); err != nil {
+			return []error{err}
+		}
+
+		// We only need to reconcile channel updates once per process startup since,
+		// configured channel settings are only loaded on startup.
+		// These operations, once completed successfully, make the condition at runStartupReconcileCheckWorker() to pass
+		// practically starting the serving of requests from the service.
+		// IMPORTANT: Everything that should run before the first request is served should happen before this
+		if err := k.connectorTypesService.ForEachConnectorCatalogEntry(k.ReconcileConnectorCatalogEntry); err != nil {
+			return []error{err}
+		}
+
+		if err := k.connectorClusterService.CleanupDeployments(); err != nil {
+			return []error{err}
+		}
+
+		k.startupReconcileDone = true
+		glog.V(5).Infoln("Catalog updates processed")
+	}
+
+	if !k.startupReconcileStopped {
+		// startup reconcile is done, stop this worker and release the worker lease for next catalog update check
+		var leaderManager *workers.LeaderElectionManager
+		k.env.MustResolve(&leaderManager)
+		workerID := k.BaseWorker.GetID()
+		go func() {
+			if err := leaderManager.StopWorker(workerID); err != nil {
+				glog.Errorf("Error stopping startup reconcile worker %s, will retry: %v", workerID, err)
+			} else {
+				glog.Infof("Stopped startup reconcile worker %s", workerID)
+				k.startupReconcileStopped = true
+			}
+		}()
```

Comment:
a comment for discussion only: 
I think the worker knowing about the leader_mgr introduce double direction dependency where the leader_mgr knows about workers and the worker knows about the leader_mgr. I am thinking maybe having a signalling mechanism where the worker send a signal that it can be stopped and for the leader_mgr to react to that signal might improve on that, what do you think?

Reply:
I agree. Maybe something like `Worker.HasTerminated()` to determine whether the worker should not be started anymore and removed from the list of workers in the leader_mgr. That will get rid of `StopWorker()` too. 
Like you said, it will simplify things quite a bit. Let me see if I can make that change easily.

Reply:
Exactly. Thanks for looking at this.

Reply:
Refactored it to a simpler implementation.